// @ionic-native/web-intent/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
import { Cordova, CordovaProperty, IonicNativePlugin, Plugin } from '@ionic-native/core';
import { Observable } from 'rxjs/Observable';
/**
 * @name Web Intent
 * @description
 * This Plugin provides a general purpose shim layer for the Android intent mechanism, exposing various ways to handle sending and receiving intents.
 * @usage
 * For usage information please refer to the plugin's Github repo.
 *
 * ```typescript
 * import { WebIntent } from '@ionic-native/web-intent';
 *
 * constructor(private webIntent: WebIntent) { }
 *
 * ...
 *
 * const options = {
 *   action: this.webIntent.ACTION_VIEW,
 *   url: 'path/to/file',
 *   type: 'application/vnd.android.package-archive'
 * };
 *
 * this.webIntent.startActivity(options).then(onSuccess, onError);
 *
 * ```
 * @interfaces
 * IntentOptions
 */
var WebIntent = (function (_super) {
    __extends(WebIntent, _super);
    function WebIntent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Launches an Android intent
     * @param options {IntentOptions}
     * @returns {Promise<any>}
     */
    /**
       * Launches an Android intent
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.startActivity = /**
       * Launches an Android intent
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    function (options) {
        return;
    };
    /**
     * Starts a new activity and return the result to the application
     * @param options {IntentOptions}
     * @returns {Promise<any>}
     */
    /**
       * Starts a new activity and return the result to the application
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.startActivityForResult = /**
       * Starts a new activity and return the result to the application
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    function (options) {
        return;
    };
    /**
     * Checks if this app was invoked with specified extra
     * @param extra {string}
     * @returns {Promise<any>}
     */
    /**
       * Checks if this app was invoked with specified extra
       * @param extra {string}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.hasExtra = /**
       * Checks if this app was invoked with specified extra
       * @param extra {string}
       * @returns {Promise<any>}
       */
    function (extra) {
        return;
    };
    /**
     * Gets the extra that this app was invoked with
     * @param extra {string}
     * @returns {Promise<any>}
     */
    /**
       * Gets the extra that this app was invoked with
       * @param extra {string}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.getExtra = /**
       * Gets the extra that this app was invoked with
       * @param extra {string}
       * @returns {Promise<any>}
       */
    function (extra) {
        return;
    };
    /**
     * Gets the Uri the app was invoked with
     * @returns {Promise<any>}
     */
    /**
       * Gets the Uri the app was invoked with
       * @returns {Promise<any>}
       */
    WebIntent.prototype.getUri = /**
       * Gets the Uri the app was invoked with
       * @returns {Promise<any>}
       */
    function () {
        return;
    };
    /**
     * Returns the content of the intent used whenever the application activity is launched
     * @returns {Observable<string>}
     */
    /**
       * Returns the content of the intent used whenever the application activity is launched
       * @returns {Observable<string>}
       */
    WebIntent.prototype.onIntent = /**
       * Returns the content of the intent used whenever the application activity is launched
       * @returns {Observable<string>}
       */
    function () {
        return;
    };
    /**
     * Sends a custom intent passing optional extras
     * @param options {IntentOptions}
     * @returns {Promise<any>}
     */
    /**
       * Sends a custom intent passing optional extras
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.sendBroadcast = /**
       * Sends a custom intent passing optional extras
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    function (options) {
        return;
    };
    /**
     * Request that a given application service be started
     * @param options {IntentOptions}
     * @returns {Promise<any>}
     */
    /**
       * Request that a given application service be started
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    WebIntent.prototype.startService = /**
       * Request that a given application service be started
       * @param options {IntentOptions}
       * @returns {Promise<any>}
       */
    function (options) {
        return;
    };
    /**
     * Registers a broadcast receiver for the specified filters
     * @param filters {any}
     * @returns {Observable<any>}
     */
    /**
       * Registers a broadcast receiver for the specified filters
       * @param filters {any}
       * @returns {Observable<any>}
       */
    WebIntent.prototype.registerBroadcastReceiver = /**
       * Registers a broadcast receiver for the specified filters
       * @param filters {any}
       * @returns {Observable<any>}
       */
    function (filters) {
        return;
    };
    /**
     * Unregisters a broadcast receiver
     */
    /**
       * Unregisters a broadcast receiver
       */
    WebIntent.prototype.unregisterBroadcastReceiver = /**
       * Unregisters a broadcast receiver
       */
    function () { };
    /**
     * @returns {Promise<any>}
     */
    /**
       * @returns {Promise<any>}
       */
    WebIntent.prototype.getIntent = /**
       * @returns {Promise<any>}
       */
    function () {
        return;
    };
    /**
     * Send a result back to the Intent that started this Activity.
     * The data can be passed using 'extras'.
     * @returns {Promise<any>}
     */
    /**
       * Send a result back to the Intent that started this Activity.
       * The data can be passed using 'extras'.
       * @returns {Promise<any>}
       */
    WebIntent.prototype.sendResult = /**
       * Send a result back to the Intent that started this Activity.
       * The data can be passed using 'extras'.
       * @returns {Promise<any>}
       */
    function (_a) {
        var _b = _a.extras;
        return;
    };
    WebIntent.decorators = [
        { type: Injectable },
    ];
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_SEND", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_VIEW", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "EXTRA_TEXT", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "EXTRA_SUBJECT", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "EXTRA_STREAM", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "EXTRA_EMAIL", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_CALL", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_SENDTO", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_GET_CONTENT", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_PICK", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_INSTALL_PACKAGE", void 0);
    __decorate([
        CordovaProperty,
        __metadata("design:type", String)
    ], WebIntent.prototype, "ACTION_UNINSTALL_PACKAGE", void 0);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "startActivity", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "startActivityForResult", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "hasExtra", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "getExtra", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "getUri", null);
    __decorate([
        Cordova({
            observable: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Observable)
    ], WebIntent.prototype, "onIntent", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "sendBroadcast", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "startService", null);
    __decorate([
        Cordova({
            observable: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Observable)
    ], WebIntent.prototype, "registerBroadcastReceiver", null);
    __decorate([
        Cordova({ sync: true }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], WebIntent.prototype, "unregisterBroadcastReceiver", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "getIntent", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], WebIntent.prototype, "sendResult", null);
    /**
     * @name Web Intent
     * @description
     * This Plugin provides a general purpose shim layer for the Android intent mechanism, exposing various ways to handle sending and receiving intents.
     * @usage
     * For usage information please refer to the plugin's Github repo.
     *
     * ```typescript
     * import { WebIntent } from '@ionic-native/web-intent';
     *
     * constructor(private webIntent: WebIntent) { }
     *
     * ...
     *
     * const options = {
     *   action: this.webIntent.ACTION_VIEW,
     *   url: 'path/to/file',
     *   type: 'application/vnd.android.package-archive'
     * };
     *
     * this.webIntent.startActivity(options).then(onSuccess, onError);
     *
     * ```
     * @interfaces
     * IntentOptions
     */
    WebIntent = __decorate([
        Plugin({
            pluginName: 'WebIntent',
            plugin: 'com-darryncampbell-cordova-plugin-intent',
            pluginRef: 'plugins.intentShim',
            repo: 'https://github.com/darryncampbell/darryncampbell-cordova-plugin-intent',
            platforms: ['Android']
        })
    ], WebIntent);
    return WebIntent;
}(IonicNativePlugin));
export { WebIntent };
